File: Easy_Level_using_Python/Array and String/242. Valid Anagram.py
# Problem: 242. Valid Anagram
# Link: https://leetcode.com/problems/valid-anagram/description/
# Time Complexity: O(n)
# Space Complexity: O(1)
# Topics: Hash Table, String, Sorting
# Solution:
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s) != len(t)):
            return False
        count = [0] * 26
        for c in s:
            count[ord(c) - ord('a')] += 1
        for c in t:
            count[ord(c) - ord('a')] -= 1

        return all(c == 0 for c in count)

File: Easy_Level_using_Cpp/Array and String/242. Valid Anagram.cpp
// Problem: 242. Valid Anagram
// Link: https://leetcode.com/problems/valid-anagram/description/
// Time Complexity: O(n)
// Space Complexity: O(1)
// Topics: Hash Table, String, Sorting
// Solution: 
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()){
            return false;
        }

        vector<int> count(26, 0);

        for(int i = 0; i < s.length(); i++){
            count[s[i] - 'a']++;
            count[t[i] - 'a']--;
        }

        for(int c: count){
            if(c != 0){
                return false;
            }
        }
        return true;
    }
};